tests/unit/test_mock_plan.py

```diff
@@ -0,0 +1,32 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import warnings
+
+import numpy as np
+import pytest
+
+from snowflake.snowpark import Session
+
+
+@pytest.fixture()
+def snowpark_session() -> Session:
+    return Session.builder.config("local_testing", True).create()
+
+
+def test_pandas_error(snowpark_session: Session):
```

Comment:
Note that this only really fails in newer versions of pandas higher than 2.2 on newer version of python on which 2.2. can actually install. We may need to change tox to iterate on more versions in order to actually capture the failure with the previous version.